tests/upload_files.rs

```diff
@@ -268,6 +268,161 @@ fn uploading_files_to_allowed_dir_works(
     Ok(())
 }
 
+#[rstest]
+#[case(server(&["-u"]))]
+#[case(server(&["-u", "-o", "error"]))]
+#[case(server(&["-u", "--on-duplicate-files", "error"]))]
+fn uploading_duplicate_file_is_prevented(#[case] server: TestServer) -> Result<(), Error> {
+    let test_file_name = "duplicate test file.txt";
+    let test_file_contents = "Test File Contents";
+    let test_file_contents_new = "New Uploaded Test File Contents";
+
+    // create the file
+    let test_file_path = server.path().join(test_file_name);
+    let _ = std::fs::write(&test_file_path, test_file_contents);
```

Comment:
Might be nicer to do:
```suggestion
    std::fs::write(&test_file_path, test_file_contents)?;
```
here and in other tests.